client/components/tinymce/plugins/contact-form/dialog.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Shortcode from 'lib/shortcode';
+import Dialog from 'components/dialog';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormButton from 'components/forms/form-button';
+
+/**
+ * object constants
+ */
+const fieldTypes = {
+	name: 'name',
+	email: 'email',
+	url: 'url',
+	checkbox: 'checkbox',
+	dropdown: 'dropdown',
+	radio: 'radio',
+	text: 'text',
+	textarea: 'textarea',
+	website: 'website'
+};
+
+const defaultForm = [
+	{ label: 'Name', type: fieldTypes.name, required: true },
+	{ label: 'Email', type: fieldTypes.email, required: true },
+	{ label: 'Website', type: fieldTypes.url },
+	{ label: 'Comment', type: fieldTypes.textarea, required: true }
+];
+
+export default React.createClass( {
+	displayName: 'ContactFormDialog',
+
+	getInitialState() {
+		return {
+			isVisible: true
+		};
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.visible && ! this.state.isVisible ) {
+			this.setState( { isVisible: true } );
+		}
+	},
+
+	render() {
+		const buttons = [
+			<FormButton
+				key="save"
+				onClick={ () => {
```

Comment:
Seems like this might be temporary anyways, but as-is, I think this would be better to reference as a function on the React class than to declare inline like you have here.